vortex-serde/src/layouts/read/mod.rs

```diff
@@ -52,25 +58,33 @@ pub type Message = (MessageId, ByteRange);
 #[derive(Debug)]
 pub enum ReadResult {
     ReadMore(Vec<Message>),
+    Selector(RowSelector),
     Batch(Array),
```

Comment:
This is jank, we should return RangedArray and convert it to RowSelectors at the top